src/firefox/index.js

```diff
@@ -249,9 +250,51 @@ export async function useProfile(
     app,
     configureThisProfile = configureProfile,
     customPrefs = {},
+    searchProfilesPath = '',
   }: UseProfileParams = {},
 ): Promise<FirefoxProfile> {
-  const profile = new FirefoxProfile({destinationDirectory: profilePath});
+  let profile;
+  const finder = new FirefoxProfile.Finder(searchProfilesPath);
```

Comment:
Given how we are using and testing the `Finder` class, I'm wondering if it would be better to create an `createProfileFinder` helper which creates the profile finder and wraps the methods that we need with promisify, something like

```js
function defaultCreateProfileFinder() {
  const finder = new FirefoxProfile.Finder();
  const readProfiles = promisify(finder.readProfiles, finder);
  return {
    getPath: promisify(finder.getPath, finder),
    hasProfileName: async () => {
       await finder.readProfiles();
       return finder.profiles.filter(
          (profileDef) => profileDef.Name === profileName
       ).length !== 0;
    }
  }
}
```

So that we can simplify `useProfile`:

```js
export async function useProfile(
   profilePath: string,
   {
      ...
      createProfileFinder: defaultCreateProfileFinder,
   }: UseProfileParams = P{,
): Promise<FirefoxProfile> {
   const finder = createProfileFinder();

   ...
}
```

and more easily test its behavior, e.g. because we can override `createProfileFinder` with a function that returns a "fake" version of our profile finder wrapper (and so we dont' even need to create the real "profiles.ini" file or the fake profiles directory in tmpDir).